packages/core/saved-objects/core-saved-objects-server/model_versions.md

```diff
@@ -0,0 +1,529 @@
+
+In that case, the new version needs to do the following:
+- add a `mappings_addition` type change to define the new mappings
+- updates the root `mappings` accordingly
+- add an updated `forwardCompatibility` as we did for the previous example
+
+The new version definition would look like: 
+
+```ts
+let modelVersion2: SavedObjectsModelVersion = {
+  // add a change defining the mapping for the new field
+  changes: [
+    {
+      type: 'mappings_addition',
+      addedMappings: {
+        dolly: { type: 'text' },
+      },
+    },
+  ],
+  schemas: {
+    // adding the new field to the forwardCompatibility schema
+    forwardCompatibility: schema.object(
+      { foo: schema.string(), bar: schema.string(), dolly: schema.string() },
+      { unknowns: 'ignore' }
+    ),
+  },
+};
+```
+
+As said, we will also need to update the root mappings definition:
+
+```ts
+mappings: {
+  properties: {
+    foo: { type: 'text' },
+    bar: { type: 'text' },
+  },
+},
+```
+
+the full type definition after the addition of the model version 2 would be:
+
+```ts
+const myType: SavedObjectsType = {
+  name: 'test',
+  namespaceType: 'single',
+  switchToModelVersionAt: '8.10.0',
+  modelVersions: {
+    1: {
+      changes: [
+        {
+          type: 'mappings_addition',
+          addedMappings: {
+            foo: { type: 'text' },
+            bar: { type: 'text' },
+          },
+        },
+      ],
+      schemas: {
+        forwardCompatibility: schema.object(
+          { foo: schema.string(), bar: schema.string() },
+          { unknowns: 'ignore' }
+        ),
+      },
+    },
+    2: {
+      changes: [
+        {
+          type: 'mappings_addition',
+          addedMappings: {
+            dolly: { type: 'text' },
+          },
+        },
+      ],
+      schemas: {
+        forwardCompatibility: schema.object(
+          { foo: schema.string(), bar: schema.string(), dolly: schema.string() },
+          { unknowns: 'ignore' }
+        ),
+      },
+    },
+  },
+  mappings: {
+    properties: {
+      foo: { type: 'text' },
+      bar: { type: 'text' },
+      dolly: { type: 'text' },
+    },
+  },
+};
+```
+
+### Adding an indexed field with a default value
+
+Now a slightly different scenario where we'd like to populate the newly introduced field with a default value.
+
+In that case, we'd need to add an additional `data_backfill` change to populate the new field's value
+(in addition to the `mappings_addition` one):
+
+```ts
+let modelVersion2: SavedObjectsModelVersion = {
+  changes: [
+    // setting the `dolly` field's default value.
+    {
+      type: 'data_backfill',
+      transform: (document) => {
+        document.attributes.dolly = 'default_value';
+        return { document };
+      },
+    },
+    // define the mappings for the new field
+    {
+      type: 'mappings_addition',
+      addedMappings: {
+        dolly: { type: 'text' },
+      },
+    },
+  ],
```

Comment:
From what I understand, the order in which changes added to this array has no effect but we chose an array so you can specify multiple of the same kind of `SavedObjectsModelChange`. I anticipate there'll be questions about that from users of the API.

WDYT about calling that out here (in this doc) and in the TS doc of `changes?

Of course my understanding might be wrong.